lib/cisco_node_utils/bgp_af.rb

```diff
@@ -98,21 +98,41 @@ def set_args_keys(hash={}) # rubocop:disable Style/AccessorMethodName
     #                      PROPERTIES                      #
     ########################################################
 
+    # IOS XR?
+    def ios_xr?
+      platform == :ios_xr
+    end
+
+    def vrf_unsupported_xr?
+      ios_xr? && @vrf != 'default'
+    end
+
+    def fail_vrf_unsupported_xr(reason)
+      fail Cisco::UnsupportedError.new(
+        'bgp_af', reason, 'set',
+        "#{reason} is not configurable on a \
+         per-VRF basis on IOS XR") if vrf_unsupported_xr?
+    end
+
     #
     # Client to client (Getter/Setter/Default)
     #
     def client_to_client
+      return nil if vrf_unsupported_xr?
       state = config_get('bgp_af', 'client_to_client', @get_args)
       state ? true : false
     end
 
     def client_to_client=(state)
+      fail_vrf_unsupported_xr('client-to-client')
+      state = !state if ios_xr?
```

Comment:
Can you add a comment that explains this line?  Not sure why it needs to be inversed.